src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,218 @@
+                        System.out.println("       " + taskList.get(taskIndex));
+                        System.out.println(dashLine);
+                    } else {
+                        System.out.println("Invalid task index.");
+                    }
+                } else if (echo.startsWith("delete")) {
+                    int taskIndex = Integer.parseInt(echo.substring(7).trim()) - 1;
+                    if (taskIndex >= 0 && taskIndex < taskList.size()) {
+                        Task removedTask = taskList.remove(taskIndex);
+                        System.out.println(dashLine);
+                        System.out.println("     Noted. I've removed this task:");
+                        System.out.println("       " + removedTask);
+                        System.out.println("     Now you have " + taskList.size() + " tasks in the list.");
+                        System.out.println(dashLine);
+                    } else {
+                        System.out.println("Invalid task index.");
+                    }
+                } else {
+                    throw new DukeException("I'm sorry, but I don't know what that means :-(");
+                }
+            } catch (DukeException e) {
+                System.out.println(dashLine);
+                System.out.println("    OOPS " + e.getMessage());
+                System.out.println(dashLine);
+            }
+        }
+
+        scanner.close();
+
+    }
+
+    public static void appendTasksToFile(ArrayList<Task> taskList, String filePath)
+            throws IOException {
+        FileWriter fw = new FileWriter(filePath);
+        for (Task task: taskList) {
+            fw.write(task.toFileString() + System.lineSeparator());
+        }
+        fw.close();
+    }
+
+    public static void loadPreviousTasks(String filePath) throws FileNotFoundException {
+        File f = new File(filePath);
+        Scanner s = new Scanner(f);
+        while (s.hasNext()) {
+            String line = s.nextLine();
+            try {
+                Task task = parseTaskFromLine(line);
+                taskList.add(task);
+            } catch (DukeException e) {
+                System.err.println("Error parsing task from file: " + e.getMessage());
+            }
+
+        }
+        s.close();
+    }
+
+    public static Task parseTaskFromLine(String line) throws DukeException {
+        String[] parts = line.split(" \\| ");
+        if (parts.length < 3) {
+            throw new DukeException("Invalid data format in the file.");
+        }
+        char taskType = parts[0].charAt(0);
+        boolean isDone = parts[1].equals("1");
+        String description = parts[2];
+
+        switch (taskType) {
+            case 'T':
+                ToDo toDoTask = new ToDo(description, taskType);
+                if (isDone) {
+                    toDoTask.markAsDone();
+                } else {
+                    toDoTask.markAsNotDone();
+                }
+                return toDoTask;
+
```

Comment:
```suggestion
        switch (taskType) {
        case 'T':
```
If i'm not wrong, the 'case' line should not have indentation